### PR TITLE
 when maxBarSize is small, each bar should only be divided by realBar…

### PR DESCRIPTION
**Pull request by BOZ88**

### CODE REVIEW

File: src/util/ChartUtils.js [hunk 1]
```diff
@@ -245,16 +245,22 @@ export const getBarPosition = ({ barGap, barCategoryGap, bandSize, sizeList = []
       originalSize >>= 0;
     }
     const size = (maxBarSize === +maxBarSize) ? Math.min(originalSize, maxBarSize) : originalSize;
+    // when maxBarSize is small, each bar should only be divided by realBarGap, not wide empty space
+    let sum = sizeList.reduce((res, entry) => (res + size), 0);
```
Comment (xile611) [inline src/util/ChartUtils.js]: It seems no `size` in `sizeList`.

Comment (BOZ88) [reply src/util/ChartUtils.js]: yes, I made a mistake. should change to 
    
const sum = len * size + (len - 1) * barGap;

**Comment by BOZ88** [comment]
fix sum calculation.

File: src/util/ChartUtils.js [hunk 2]
```diff
@@ -245,16 +245,21 @@ export const getBarPosition = ({ barGap, barCategoryGap, bandSize, sizeList = []
       originalSize >>= 0;
     }
     const size = (maxBarSize === +maxBarSize) ? Math.min(originalSize, maxBarSize) : originalSize;
+    // when maxBarSize is small, each bar should only be divided by realBarGap, not wide empty space
+    const sum = len * size + (len - 1) * barGap;
```
Comment (xile611) [inline src/util/ChartUtils.js]: `bargap` can be a percent ...

**Pull request closed without merging**